src/github/SyftGithubAction.ts

```diff
@@ -0,0 +1,398 @@
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as github from "@actions/github";
+import * as cache from "@actions/tool-cache";
+import {
+  PullRequestEvent,
+  Release,
+  ReleaseEvent,
+} from "@octokit/webhooks-types";
+import * as fs from "fs";
+import * as os from "os";
+import path from "path";
+import stream from "stream";
+import { SyftOptions } from "../Syft";
+import { dashWrap, getClient } from "./GithubClient";
+
+export const SYFT_BINARY_NAME = "syft";
+export const SYFT_VERSION = "v0.21.0";
+
+const PRIOR_ARTIFACT_ENV_VAR = "ANCHORE_SBOM_ACTION_PRIOR_ARTIFACT";
+
+/**
+ * Tries to get a unique artifact name or otherwise as appropriate as possible
+ */
+function getArtifactName(): string {
+  const fileName = core.getInput("artifact_name");
+
+  // if there is an explicit filename just return it, this could cause issues
+  // where earlier sboms are overwritten by later ones
+  if (fileName) {
+    return fileName;
+  }
+
+  const { job, action } = github.context;
+  // when run without an id, we get various auto-generated names, like:
+  // __self __self_2 __anchore_sbom-action  __anchore_sbom-action_2 etc.
+  // so just keep the number at the end if there is one, otherwise
+  // this will not match an id unless for some reason it starts with __
+  let stepName = action.replace(/__[-_a-z]+/, "");
+  if (stepName) {
+    stepName = `-${stepName}`;
+  }
+  const format = getSbomFormat();
+  return `sbom-${job}${stepName}.${format}`;
+}
+
+/**
+ * Gets a reference to the syft command and executes the syft action
+ * @param input syft input parameters
+ * @param format syft output format
+ */
+async function executeSyft({ input, format }: SyftOptions): Promise<string> {
```

Comment:
nit: This function is a bit large. I had a bit of trouble parsing the `try`/`catch` section in particular, but I did end up figuring out what was happening and why. Not a firm request, but if there's any way this can be decomposed into smaller-scoped implementations, I think that'd help me.

Reply:
Agreed, the main purpose was to make sure the syft output gets dumped to the logs, but I think it's probably not necessary so simplified the exception handling. I don't think I should make this function much smaller, it's basically responsible for constructing the command line parameters and capturing the output of syft. WDYT?

Reply:
I think the error-handling changes helped, thanks for that. I agree that it's mostly good at this point.

It looks like you might be able to factor out the work of "prepare `args` using `SyftOptions`".

E.g.:

```ts
const args = determineArgs({ input, format });
```

and

```ts
function determineArgs({ input, format }: SyftOptions): string[] {
  let args = ["packages"];

  if ("image" in input && input.image) {
    args = [...args, `docker:${input.image}`];
  } else if ("path" in input && input.path) {
    args = [...args, `dir:${input.path}`];
  } else {
    throw new Error("Invalid input, no image or path specified");
  }

  args = [...args, "-o", format];
  return args;
}
```

But if that's not helpful here, **please disregard**. This function is already in a state I'm comfortable with! 👍